### ISSUE TITLE
implicit-arrow-linebreak doesn't enforce closing paren location

### ISSUE THREAD

**Issue by willlma** [eslint/eslint#12448]
**Tell us about your environment**

* **ESLint Version:**
v6.5.1
* **Node Version:**
v8.12.0
* **npm Version:**
v6.9.0 (but I'm using yarn v1.15.2 for this project)
**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint@10.0.3
**Please show your full configuration:**
<details>
<summary>Configuration</summary>
Using eslint-config-airbnb@18.0.1

```yml
extends: airbnb
parser: babel-eslint
env:
  browser: true
rules:
  comma-dangle: [warn, always-multiline, {functions: never}]
  import/prefer-default-export: 0
  jsx-quotes: [warn, prefer-single]
  max-len: [warn, { code: 80 }]
  radix: [warn, as-needed]
  react/destructuring-assignment: [warn, always, { ignoreClassFields: true }]
  react/jsx-filename-extension: 0
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const doSomeFoo = (foo) => (
  foo);
```

```bash
node_modules/.bin/eslint app/javascript/specifications/actions/form/index.js
```

**What did you expect to happen?**
A warning telling me that my closing paren should be on a new line, like in [the docs](https://eslint.org/docs/rules/implicit-arrow-linebreak):
```
const doSomeFoo = (foo) => (
  foo
);
```

**What actually happened? Please include the actual, raw output from ESLint.**
No output

**Are you willing to submit a pull request to fix this bug?**
I have never worked on this project, so unless someone were to point me in the right direction, likely not.

**Comment by g-plane** [comment]
Thanks for this issue.

I think this is expected, because the parens are parts of your arrow function body, and this rule only cares the first token of the body, not the parens.

**Comment by willlma** [comment]
Thanks for the response. Is there another way to enforce the location of the closing paren?

**Comment by kaicataldo** [comment]
The parentheses here don't really have anything to do with the arrow function expression, so I don't think it should be enforced with an arrow function expression rule. I don't think we currently have a generic rule for enforcing newlines within parentheses, but please feel free to create a [new rule proposal](https://github.com/eslint/eslint/issues/new?assignees=&labels=triage%2C+rule%2C+feature&template=NEW_RULE.md&title=) if you'd like to advocate for this!

**Comment by g-plane** [comment]
@willlma I'm considering implementing this as a separated plugin.

**Comment by g-plane** [comment]
@willlma I've implemented this in a separated plugin. You can try: https://github.com/g-plane/eslint-plugin-mayaka/blob/master/docs/arrow-body-parens-linebreak.md .

**Comment by willlma** [comment]
Many thanks!
